src/helpers/bookmarkHelper.ts

```diff
@@ -0,0 +1,67 @@
+import { QUERY_TYPES } from '../queries';
+import { addToStore, readFromStore } from './storageHelper';
+
+export const BOOKMARKS_STORE_KEY = 'BOOKMARKED_ITEMS';
+
+export type BookmarkList = {
+  [key: string]: string[];
+};
+
+export const getListQueryType = (singleItemType: string) => {
+  switch (singleItemType) {
+    case QUERY_TYPES.EVENT_RECORD:
+      return QUERY_TYPES.EVENT_RECORDS;
+    case QUERY_TYPES.NEWS_ITEM:
+      return QUERY_TYPES.NEWS_ITEMS;
+    case QUERY_TYPES.TOUR:
+      return QUERY_TYPES.TOURS;
+    case QUERY_TYPES.POINT_OF_INTEREST:
+      return QUERY_TYPES.POINTS_OF_INTEREST;
+    default:
+      return singleItemType;
+  }
+};
+
+// add category as a suffix to the itemType to avoid handling NewsItem different
+// than the other item types in every other component
+// this way the hierarchy in the storage remains flat and homogeneous
+export const getKeyFromTypeAndCategory = (itemType: string, category?: number) =>
+  category ? `${itemType}-${category}` : itemType;
+
+export const toggleBookmark = async (itemType: string, id: string, category?: number) => {
+  const bookmarks: BookmarkList | undefined = await readFromStore(BOOKMARKS_STORE_KEY);
+  const listQueryType = getListQueryType(itemType);
+
+  // FIXME: add error handling
```

Comment:
what error handling is being thought of here?

Reply:
once we have some sort of error reporting, we can report this case. until then we can't do much about it.